libraries/AP_NavEKF2/AP_NavEKF2_Measurements.cpp

```diff
@@ -374,6 +374,19 @@ bool NavEKF2_core::readDeltaVelocity(uint8_t ins_index, Vector3f &dVel, float &d
 *             Global Position Measurement               *
 ********************************************************/
 
+/*
+  propogate forward the position data in a GPS measurement to the
+  given time.
+ */
+void NavEKF2_core::propogateGpsData(gps_elements &gpsData, uint32_t new_time_ms)
+{
+    float dtime = (new_time_ms - gpsData.time_ms) * 1.0e-3f;
+    gpsData.pos.x += dtime * gpsData.vel.x;
```

Comment:
Isn't the EKF already accounting for this for you when it rewinds the data? As I understand it this should only be needed if the GPS data is more delayed then the IMU history. (Highly possible I'm missing something about the EKF here)